src/braket/circuits/angled_gate.py

```diff
@@ -41,22 +48,30 @@ def __init__(self, angle: float, qubit_count: Optional[int], ascii_symbols: Sequ
         super().__init__(qubit_count=qubit_count, ascii_symbols=ascii_symbols)
         if angle is None:
             raise ValueError("angle must not be None")
-        self._angle = float(angle)  # explicit casting in case angle is e.g. np.float32
+        if isinstance(angle, FreeParameter):
+            self._angle = angle
+            self._parameter = angle
+        else:
+            self._angle = float(angle)  # explicit casting in case angle is e.g. np.float32
+            self._parameter = None
 
     @property
-    def angle(self) -> float:
+    def angle(self) -> Union[FreeParameter, float]:
         """
         Returns the angle for the gate
 
         Returns:
-            angle (float): The angle of the gate in radians
+            angle (Union[FreeParameter, float]): The angle of the gate in radians
```

Comment:
Returns aren't named:
```suggestion
            Union[FreeParameter, float]: The angle of the gate in radians
```